# To inform the algorithm to exit automatically when no data is received
autoexit:           1

# Declare the lidar topics to merge
lidar_topic:        [
                      "/livox/lidar_ouster"
                    ]

merged_lidar_topic: "/merged_pc"

# Lidar extrinsics: Transform of coordinates in lidar's frame to body
# NOTE TO USER: The (4,4) entry of the extrinsics is used to declare
# whether stamp is at beginning (1.000) or end (0.000) of scan
lidar_extr:         [
                      # Livox
                      1, 0, 0, -0.01100,
                      0, 1, 0, -0.02329,
                      0, 0, 1,  0.04412,
                      0, 0, 0,  1.00000
                    ]

# Minimum lidar range to admit
min_range:          0.25

# IMU topic
imu_topic:          "/livox/imu"

# Imu data will be multiplied with this to get m/s^2
imu_scale:          9.82

# Gravity constant
GRAV:               9.82

# IMU noises
GYR_N:              1.0e-3
GYR_W:              1.0e-6
ACC_N:              1.0e-2
ACC_W:              1.0e-5

# Weightage of a basic lidar factor
lidar_weight:       2.0

# Number of packets to process at time
WINDOW_SIZE:        6
# Number of substates to give to each packet
N_SUB_SEG:          2

# Downsampling grid size, chose -1 for no downsampling
leaf_size:          0.05          # Downsample scale of input pointcloud, also the scale of ufomap leaf
assoc_spacing:      0.8
surfel_map_depth:   16
surfel_min_point:   8
surfel_min_depth:   1
surfel_query_depth: 7             # Equal to Dmax + 1. Dmax is defined in the paper
surfel_intsect_rad: 0.1           # r in the the paper
surfel_min_plnrty:  0.7

# Distance to surfel and min score thresholds
dis_to_surfel_max:  0.5
score_min:          0.0

# Downsample coefficient of lidar scans, ds_rate = n means keep every n-th point.
ds_rate:            [
                      1
                    ]

# Number of pointclouds to merge into one
sweep_len:          2

# Threshold to add new keyframe
kf_min_dis:         1.0
kf_min_angle:       10.0

# Optimization parameters
lidar_loss_thres:   10.0

# Solver:
# Trust region strategy: "lm", "dogleg"
trustRegType:       "lm"
# Linear Algebra library used: "eigen", "lapack", "cuda"
linAlgbLib:         "cuda" 
max_solve_time:     0.3
max_iterations:     40
ensure_real_time:   0

# Sensors fused
fuse_lidar:         1
fuse_imu:           1

regularize_imu:     1
imu_init_time:      1.0
max_outer_iters:    3
max_lidar_factor:   8000   # Maximum number of lidar factors

# Loop closure parameters
loop_en:            1
loop_kf_nbr:        20    # Number of neighbour to check for loop closure
loop_kfid_mindiff:  20    # Only check for loop when keyframes have this much difference

icpMaxIter:         100   # Maximum iterations for ICP
icpFitnessThres:    0.6   # Fitness threshold for ICP check
histDis:            15    # Maximum correspondence distance for icp

# Bundle adjustment params
rib_edge:           100
odom_q_noise:       0.1
odom_p_noise:       0.1
loop_weight:        0.05

# Number of optimizations before quitting
debug_exit:        -1
publish_map:        0

# Log directory
log_dir:            "/home/tmn/slict_log/atv"